File: experiment_utils/workloads/sha_scripts/sha.c
#include <openssl/sha.h>
#include <stdio.h>
#include <string.h>
#include <unistd.h>

int main() {
    // 초기화할 메시지
    unsigned char message[] = "Hello, World!";
    
    while(1){
        // SHA-256 해시를 저장할 배열
        unsigned char hash[SHA256_DIGEST_LENGTH];
        
        // SHA-256 컨텍스트 초기화
        SHA256_CTX sha256;
        SHA256_Init(&sha256);
        
        // 해시 계산
        SHA256_Update(&sha256, message, strlen((char*)message));
        SHA256_Final(hash, &sha256);
        
        // 해시 출력
        int i;
        for (i = 0; i < SHA256_DIGEST_LENGTH; i++) {
            printf("%02x", hash[i]);
        }
        printf("\n");

        sleep(3);
    }

    return 0;
}

File: workload_instruction_analyzer/bytecode_tracking/exp_workloads/sha/sha.py
from sha import sha256_hash
import time

# 해시할 메시지
message = b"Hello, World!"

while(True):
    # sha256_hash 함수를 호출하여 해시 값을 얻음
    hash_value = sha256_hash(message)
    
    # 해시 값을 16진수 문자열로 변환하여 출력
    print(f"SHA-256 hash of '{message.decode()}': {hash_value.hex()}")

    time.sleep(3)
